.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,31 @@
+name: gh-pages
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for specific branches
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set-up Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: "10.x"
+      - run: npm install
+      - run: npm run build
+      - run: npm run test
+      - run: npm run lint
+      - name: Deploy
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```

Comment:
In the `on` section, we trigger workflow on a PR to master to make sure that PRs contain code that passes tests and lint. However, we don't want such code to deploy to production (in our case, to Github Pages)! It may not be ready and it hasn't passed the review. This is why we add a condition for the `Deploy` step - now the deploys step will run only when we push to `master` :)